round4/team-10.md

```diff
@@ -0,0 +1,211 @@
+
+https://zapier.com/blog/productivity-extensions-for-chrome/
+https://old.reddit.com/r/chrome/comments/18vzy7g/what_are_your_favorite_productivity_extensions/
+https://old.reddit.com/r/chrome_extensions/comments/15xg1vv/extensions_for_productivity/
+[Tab for a cause](https://tab.gladly.io/)
+[Freedom](https://addons.mozilla.org/en-US/firefox/addon/freedom-website-blocker/)
+
+## Technical Details
+
+### User Interface
+
+_[Describe the user interface of your Chrome Extension: What will users see when they interact with your extension? How will they interact with the extension? You can include rough sketches or wireframes to illustrate your design, but name your files `<project id>-ui-1.png`, `<project id>-ui-2.png`, etc.]_
+
+[View POC](https://github.com/rayan-yu/exTimesion/blob/main/IMG_D4ECF3952792-1.jpeg)
+
+_[Describe which Chrome UI/UX elements you will use in your extension, such as pop-ups, context menus, browser actions, omnibox, sidebar, etc.]_
+
+We would use pop-ups to show specific page statistics/breakdown, and we would allow the user to open up a larger sidebar to show the entire dashboard of statistics, if they'd like to see a more detailed breakdown.
+
+### API, Libraries, and Frameworks
+
+Tabs API (chrome.tabs): This is crucial for managing tabs within Chrome. We plan on using it to listen for tab activation, creation, update, and removal events. This way, we can start or stop the timer based on the user's activity in each tab.
+
+Windows API (chrome.windows): Similar to the Tabs API, but for managing browser windows. It's important for understanding when a user switches windows, which can affect how we track time spent in tabs.
+
+Storage API (chrome.storage): This API is essential for saving data locally on the user's machine for persistent storage. We would probably use this to store the amount of time spent on each tab, allowing the extension to maintain records between browsing sessions.
+
+chrome.idle: Detects when the user is idle, which can help in pausing time tracking when the user is not actively browsing.
+Content Scripts (chrome.contentScripts): While not directly involved in time tracking, content scripts can be used to interact with the content of a web page. For example, we’re interested in showing an overlay on the tab that's currently being timed or inject specific elements to provide feedback to the user.
+
+Chart.js: simple yet flexible JavaScript charting library that can be used to display analytics data through line charts, bar charts, etc.
+
+D3.js: more complex and powerful library for data visualization that allows for highly customizable graphics. It's useful if you need more advanced visualizations.
+
+Moment.js: now considered a legacy project in maintenance mode, it's still widely used for parsing, validating, manipulating, and displaying dates and times in JavaScript.
+
+Node.js: running JavaScript code outside of a browser.
+
+Npm: managing the project's dependencies.
+
```

Comment:
Thanks for writing down the general tech stack/technologies you'd use! It's v helpful when giving feedback--this all lgtm!